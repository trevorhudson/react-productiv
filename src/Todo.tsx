import React from "react";


interface TodoInterface {
  id: string,
  title: string,
  description: string,
  priority: string
}

interface PropsInterface  {
  todo: TodoInterface | null
}

/** Simple presentation component for a todo.
 *
 * Props:
 * - todo: like { id, title, description, priority }
 *
 * { EditableTodo, TopTodo } -> Todo
 **/


const Todo: React.FC<PropsInterface> = ({todo}) =>{
  return (
    <div className="Todo">
      <div><b>{todo!.title}</b> <small>{todo!.priority}</small></div>
      <div><small>{todo!.description}</small></div>
    </div>
  );
}

export default Todo;
