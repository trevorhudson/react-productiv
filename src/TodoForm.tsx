import React, { useState } from "react";

interface FormData {
  id: string,
  title: string,
  description: string,
  priority: string
}


interface PropsInterface {
  initialFormData: FormData;
  handleSave: (formData: FormData) => void;
}

const TodoForm: React.FC<PropsInterface> = ({initialFormData, handleSave}) => {
  const [formData, setFormData] = useState(initialFormData);

  /** Update form input. */
  function handleChange(evt: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement>) {
    const { name, value } = evt.target;
    setFormData(fData => ({
      ...fData,
      [name]: name === "priority" ? parseInt(value) : value,
    }));
  }

  /** Call parent function and clear form. */
  function handleSubmit(evt: React.FormEvent<HTMLFormElement>) {
    evt.preventDefault();
    handleSave(formData);
    setFormData(initialFormData);
  }

  return (
    <form className="NewTodoForm" onSubmit={handleSubmit}>

      <div className="mb-3">
        <input
          id="newTodo-title"
          name="title"
          className="form-control"
          placeholder="Title"
          onChange={handleChange}
          value={formData.title}
          aria-label="Title"
        />
      </div>

      <div className="mb-3">
        <textarea
          id="newTodo-description"
          name="description"
          className="form-control"
          placeholder="Description"
          onChange={handleChange}
          value={formData.description}
          aria-label="Description"
        />
      </div>

      <div className="mb-3 d-flex justify-content-between">
        <div className="w-75 d-flex justify-content-between">
          <label htmlFor="newTodo-priority"
            className="d-inline-flex">Priority:&nbsp;&nbsp;
          </label>
          <select id="newTodo-priority"
            name="priority"
            value={formData.priority}
            onChange={handleChange}
            className="form-control form-control-sm d-inline-flex"
            >
            <option value={1}>Ultra-Über</option>
            <option value={2}>Über</option>
            <option value={3}>Meh</option>
          </select>
        </div>
        <button className="btn-primary rig btn btn-sm NewTodoForm-addBtn">
          Gø!
        </button>
      </div>

    </form>
  );
}

export default TodoForm;
