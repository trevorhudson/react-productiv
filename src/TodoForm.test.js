import { render,fireEvent,getByLabelText, queryByText } from "@testing-library/react";
import Todo from './Todo.js';
import TodoForm from './TodoForm.js';
import ToDoApp from './ToDoApp.js';


describe("TodoForm component test" , function() {
  const initialFormData = {
    title:"test",
    description:"test",
    priority: "1",
  }

  it("renders", function() {
    render(<TodoForm initialFormData={initialFormData} handleSave={ToDoApp.handleSave}/>)
  });

  it('passes snapshot test', function () {
    const { container } = render(
    <TodoForm
    initialFormData={initialFormData}
    handleSave={ToDoApp.handleSave}/>);
    expect(container).toMatchSnapshot();
  });

  // it("updates on submitButton", function() {
  //   const { container } = render(
  //     <TodoForm
  //     initialFormData={initialFormData}
  //     handleSave={ToDoApp.handleSave}/>);

  //   const submitButton = container.querySelector(".NewTodoForm-addBtn");
  //   fireEvent.click(submitButton);
  //   // expect.assertions(1);
  //   expect()
  // })
  // it("updates per change", function() {
  //   const { container } = render(
  //     <TodoForm
  //     initialFormData={initialFormData}
  //     handleSave={ToDoApp.handleSave}/>);

  //     const inputField = container.querySelector("#newTodo-description");
  //     fireEvent.change(inputField,{target:{value:"test123"}})

  //     expect(container.queryByText("test123")).toBeInTheDocument();

  // })

})