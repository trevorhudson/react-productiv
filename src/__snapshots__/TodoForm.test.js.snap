// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm component test passes snapshot test 1`] = `
<div>
  <form
    class="NewTodoForm"
  >
    <div
      class="mb-3"
    >
      <input
        aria-label="Title"
        class="form-control"
        id="newTodo-title"
        name="title"
        placeholder="Title"
        value="test"
      />
    </div>
    <div
      class="mb-3"
    >
      <textarea
        aria-label="Description"
        class="form-control"
        id="newTodo-description"
        name="description"
        placeholder="Description"
      >
        test
      </textarea>
    </div>
    <div
      class="mb-3 d-flex justify-content-between"
    >
      <div
        class="w-75 d-flex justify-content-between"
      >
        <label
          class="d-inline-flex"
          for="newTodo-priority"
        >
          Priority:  
        </label>
        <select
          class="form-control form-control-sm d-inline-flex"
          id="newTodo-priority"
          name="priority"
          type="number"
        >
          <option
            value="1"
          >
            Ultra-Über
          </option>
          <option
            value="2"
          >
            Über
          </option>
          <option
            value="3"
          >
            Meh
          </option>
        </select>
      </div>
      <button
        class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
      >
        Gø!
      </button>
    </div>
  </form>
</div>
`;
